src/Product.php

```diff
@@ -355,18 +369,14 @@ public function getProductFullUrl($path, $mainVariationId)
             $prefix .= '/' . $this->getProductUrlPrefix();
         }
 
-        // Using trim just in case if path could be passed with and without forward slash
-        $path = '/' . ltrim($path, '/');
-        $path = rtrim($path, '/');
-
         return sprintf(
             '%s%s%s%s_%s_%s',
             $this->protocol,
             $this->getStoreUrl(),
             $prefix,
-            $path,
+            $this->path,
```

Comment:
It seems `getProductFullUrl` is called before path is set, does that mean it's an empty string?

Reply:
No it is not. The path is set when we initially get the data from the product. The URL is now set when we go through each variation, which is done after setting the initial data from the product.